l10n_ch_sepa_direct_debit/wizard/export_sdd.py

```diff
@@ -0,0 +1,337 @@
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+from lxml import etree
+
+
+class BankingExportSepaWizard(orm.TransientModel):
+    _inherit = 'banking.export.sdd.wizard'
+
+
+def create_sepa(self, cr, uid, ids, context=None):
+    '''
+    Creates the SEPA Direct Debit file. That's the important code !
+    '''
+    sepa_export = self.browse(cr, uid, ids[0], context=context)
+
+    pain_flavor = sepa_export.payment_order_ids[0].mode.type.code
+    convert_to_ascii = \
+        sepa_export.payment_order_ids[0].mode.convert_to_ascii
+    if pain_flavor == 'pain.008.001.02':
+        bic_xml_tag = 'BIC'
+        name_maxsize = 70
+        root_xml_tag = 'CstmrDrctDbtInitn'
+    elif pain_flavor == 'pain.008.001.03':
+        bic_xml_tag = 'BICFI'
+        name_maxsize = 140
+        root_xml_tag = 'CstmrDrctDbtInitn'
+    elif pain_flavor == 'pain.008.001.04':
+        bic_xml_tag = 'BICFI'
+        name_maxsize = 140
+        root_xml_tag = 'CstmrDrctDbtInitn'
+    elif pain_flavor == 'pain.008.001.02.ch.01':
+        bic_xml_tag = 'BICFI'
+        name_maxsize = 140
+        root_xml_tag = 'CstmrDrctDbtInitn'
+    else:
+        raise orm.except_orm(
+            _('Error:'),
+            _("Payment Type Code '%s' is not supported. The only "
+              "Payment Type Code supported for SEPA Direct Debit "
+              "are 'pain.008.001.02', 'pain.008.001.03' and "
+              "'pain.008.001.04'.") % pain_flavor)
+
+    if pain_flavor == 'pain.008.001.02.ch.01':
+        gen_args = {
+            'bic_xml_tag': bic_xml_tag,
+            'name_maxsize': name_maxsize,
+            'convert_to_ascii': convert_to_ascii,
+            'payment_method': 'DD',
+            'pain_flavor': pain_flavor,
+            'sepa_export': sepa_export,
+            'file_obj': self.pool['banking.export.sdd'],
+            'pain_xsd_file':
+                'l10n_ch_sepa_direct_debit/data/%s.xsd' % pain_flavor,
+        }
+    else:
+        gen_args = {
+            'bic_xml_tag': bic_xml_tag,
+            'name_maxsize': name_maxsize,
+            'convert_to_ascii': convert_to_ascii,
+            'payment_method': 'DD',
+            'pain_flavor': pain_flavor,
+            'sepa_export': sepa_export,
+            'file_obj': self.pool['banking.export.sdd'],
+            'pain_xsd_file':
+                'account_banking_sepa_direct_debit/data/%s.xsd' % pain_flavor,
+        }
```

Comment:
Same comment than for credit transfer. Here you could have a placeholder for module